src/bootstrap/tarball.rs

```diff
@@ -316,6 +316,20 @@ impl<'a> Tarball<'a> {
         }
         self.builder.run(&mut cmd);
 
+        // Ensure there are no symbolic links in the tarball. In particular,
+        // rustup-toolchain-install-master and most versions of Windows can't handle symbolic links.
+        // NOTE: we special-case rustc-nightly-src, which neither rustup nor RTIM actually know how to install,
+        // and does contain symlinks.
+        let decompressed_output = self.temp_dir.join(&package_name);
+        if !self.builder.config.dry_run && self.target.as_deref() != Some("src") {
```

Comment:
I'd prefer to avoid gating on self.target != src here -- can we add a parameter on the struct that's set by a method (e.g., `permit_symlinks(true)`)?